prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -2443,10 +2544,26 @@ int rxr_pkt_proc_fetch_rta(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
 
 	offset = 0;
 	for (i = 0; i < rx_entry->iov_count; ++i) {
-		ofi_atomic_readwrite_handlers[op][dt](rx_entry->iov[i].iov_base,
-						      data + offset,
-						      rx_entry->atomrsp_data + offset,
-						      rx_entry->iov[i].iov_len / dtsize);
+		efa_mr = (struct efa_mr*) ofi_mr_map_get(&ep->util_ep.domain->mr_map, (rxr_get_rta_hdr(pkt_entry->pkt)->rma_iov + i)->key);
+		hmem_iface = efa_mr->peer.iface;
+		if (hmem_iface == FI_HMEM_SYSTEM) {
+			ofi_atomic_readwrite_handlers[op][dt](rx_entry->iov[i].iov_base,
+			                                      data + offset,
+			                                      rx_entry->atomrsp_data + offset,
+			                                      rx_entry->iov[i].iov_len / dtsize);
+		}
+		else if (hmem_iface == FI_HMEM_CUDA) {
+			ret = rxr_cuda_fetch_atomic(efa_mr, &rx_entry->iov[i], data + offset,
+			                            rx_entry->atomrsp_data + offset, dtsize, op, dt);
+			if (OFI_UNLIKELY(ret != 0)) {
+				return ret;
+			}
+		}
+		else {
+			FI_WARN(&rxr_prov, FI_LOG_CQ, "Atomics not supported for FI_HMEM_IFACE: %d.\n", hmem_iface);
+			return ENOSYS;
+		}
```

Comment:
nit: I'd prefer this modeled as a `switch`...`case` construct since it's a little cleaner.

```c
switch (hmem_iface) {
case FI_HMEM_SYSTEM:
    ...
    break;
case FI_HMEM_CUDA:
    ...
    break;
default:
    FI_WARN(...);
    return ENOSYS;
}
```

Reply:
changed to if/else.